arkfilepicker/src/main/java/space/taran/arkfilepicker/folders/FoldersRepo.kt

```diff
@@ -119,11 +113,57 @@ class FoldersRepo(private val appCtx: Context) {
         } ?: listOf(fav)
         folders = mutFolders
 
-        val arkFolder = root.arkFolder()
-        require(arkFolder.exists()) { "Ark folder must exist" }
-        val favsFile = arkFolder.arkFavorites()
-        val jsonFavs = JsonFavorites(folders[root]!!.map(Path::toString))
-        favsFile.writeText(klaxon.toJsonString(jsonFavs))
+        writeFavorites(root)
+    }
+
+    suspend fun forgetRoot(root: Path) =
+        withContext(Dispatchers.IO) {
+            if (folders.containsKey(root)) {
+                val mFolders = folders.toMutableMap()
+                mFolders.remove(root)
+                folders = mFolders
+                writeRoots()
+            }
+        }
+
+    suspend fun deleteRoot(root: Path) =
+        withContext(Dispatchers.IO) {
+            if (folders.containsKey(root)) {
+                forgetRoot(root)
+                deleteFilesRecursively(root)
+            }
+        }
+
+    suspend fun forgetFavorite(root: Path, favorite: Path) =
+        withContext(Dispatchers.IO) {
+            if (folders.containsKey(root)) {
+                val favorites = folders[root]
+                val favRelative = root.relativize(favorite)
+                if (favorites != null && favorites.contains(favRelative)) {
+                    val mFavorites = favorites.toMutableList()
+                    mFavorites.remove(favRelative)
+                    folders = folders.plus(mapOf(root to mFavorites))
+                    writeFavorites(root)
+                }
+            }
+        }
+
+    suspend fun deleteFavorite(root: Path, favorite: Path) =
+        withContext(Dispatchers.IO) {
+            forgetFavorite(root, favorite)
+            deleteFilesRecursively(favorite)
+        }
+
+    private fun deleteFilesRecursively(path: Path) {
+        val files = Files.list(path)
+        if (files != null) {
+            files.forEach {
+                if (it.isDirectory())
+                    deleteFilesRecursively(path)
+                else Files.delete(it)
+            }
+        }
+        Files.delete(path)
```

Comment:
Maybe `path.toFile().deleteRecursively()`?

Reply:
Thanks for the suggestion, I think it will solve the stackoverflow